packages/gateway/src/gateway.js

```diff
@@ -77,18 +85,36 @@ export async function gatewayGet(request, env, ctx) {
     // forward winner gateway response
     return winnerGwResponse.response
   } catch (err) {
+    let responses
+    // We can already settle requests if Aggregate Error, as all promises were already rejected
+    if (err instanceof AggregateError) {
+      responses = await pSettle(gatewayReqs)
+    }
```

Comment:
What happens if some promises fulfilled but none were `response.ok` and the rest rejected? Do we still get an aggregate error?

Yes, the filter causes a `FilterError` to be thrown here: https://github.com/sindresorhus/p-some/blob/a7030ea6ad9971867ba5dfdb09034a416f3cf8e3/index.js#L60-L62

I think this will _always_ be an `AggregateError` in our case. Or in other words, all of our promises will have rejected or been filtered (effectively rejected) here. We should _always_ settle here to get the responses:

```suggestion
    // All promises will have been rejected or filtered (effectively rejected) here.
    responses = await pSettle(gatewayReqs)
```

Reply:
That's right, thanks